day_01/day_01.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+import argparse
+import pathlib
+import regex as re
+
+import day_01
+
+DEFAULT_DATA = pathlib.Path(day_01.__file__).parent.parent / "data"/ "day1.txt"
+
+
+def load(filename: str | pathlib.Path) -> list[str]:
+    with open(filename) as f:
+        return f.readlines()
+
+
+def solve_q1(data: list[str]) -> int:
+    digits: list[int] = []
+    digit_re = re.compile(r'\d')
+
+    for text in data:
+        first = digit_re.search(text).group()
+        last  = digit_re.search(text[::-1]).group()
+        digits.append(int(first + last))
+
+    return sum(digits)
+
+
+def solve_q2(data: list[str]) -> int:
+    numbers = ('one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine')
+    mapping: dict[str, int] = {}
+    for n, name in enumerate(numbers, start=1):
+        mapping[name] = n
+        mapping[str(n)] = n
+
+    pattern = r'|'.join(numbers)
+    digits_re = re.compile(rf'\d|{pattern}')
```

Comment:
Build the pattern from that list using a mixture of fstring and pattern protected r-string.